gem/spec/metasploit_payloads/metasploit_payloads_spec.rb

```diff
@@ -0,0 +1,252 @@
+# frozen_string_literal: true
+
+require 'metasploit-payloads'
+
+RSpec.describe ::MetasploitPayloads do
+  describe '::VERSION' do
+    it 'has a version number' do
+      expect(::MetasploitPayloads::VERSION).not_to be nil
+    end
+  end
+
+  describe '::Error' do
+    it 'has an Error class' do
+      expect(::MetasploitPayloads::Error.superclass).to be(::StandardError)
+    end
+
+    it 'has a NotFoundError class' do
+      expect(::MetasploitPayloads::NotFoundError.superclass).to be(::MetasploitPayloads::Error)
+    end
+
+    it 'has a NotReadableError class' do
+      expect(::MetasploitPayloads::NotReadableError.superclass).to be(::MetasploitPayloads::Error)
+    end
+  end
+
+  describe '#readable_path' do
+    before :all do
+      @sample_file = 'meterpreter/meterpreter.py'
+    end
+
+    before :each do
+      allow(::File).to receive(:exist?).and_call_original
+      allow(::File).to receive(:readable?).and_call_original
+    end
+
+    context 'when the path is not readable' do
+      it 'raises a ::MetasploitPayloads::NotReadableError' do
+        allow(::File).to receive(:exist?).with(@sample_file).and_return(true)
+        allow(::File).to receive(:readable?).with(@sample_file).and_return(false)
+
+        expect { subject.send(:readable_path, @sample_file) }.to raise_error(::MetasploitPayloads::NotReadableError)
+      end
+    end
```

Comment:
https://www.betterspecs.org/#let

```suggestion
    let(:sample_file) { 'meterpreter/meterpreter.py' }

    before :each do
      allow(::File).to receive(:exist?).and_call_original
      allow(::File).to receive(:readable?).and_call_original
    end

    context 'when the path is not readable' do
      it 'raises a ::MetasploitPayloads::NotReadableError' do
        allow(::File).to receive(:exist?).with(sample_file).and_return(true)
        allow(::File).to receive(:readable?).with(sample_file).and_return(false)
        expect { subject.readable_path(sample_file) }.to raise_error(::MetasploitPayloads::NotReadableError)
      end
    end
```